src/read.c

```diff
@@ -863,15 +864,31 @@ static avifResult avifDecoderItemMaxExtent(const avifDecoderItem * item, const a
     uint64_t maxOffset = 0;
     for (uint32_t extentIter = 0; extentIter < item->extents.count; ++extentIter) {
         avifExtent * extent = &item->extents.extent[extentIter];
-        const size_t usedExtentSize = (extent->size < remainingBytes) ? extent->size : remainingBytes;
+
+        // Make local copies of extent->offset and extent->size as they might need to be adjusted
+        // due to the sample's offset.
+        uint64_t startOffset = extent->offset;
+        uint64_t extentSize = extent->size;
```

Comment:
Declare this as size_t because `extent->size` is size_t:

`size_t extentSize = extent->size;`